Kernel/Arch/aarch64/ASM_wrapper.h

```diff
@@ -91,6 +91,32 @@ inline void enter_el1_from_el2()
                      : "x0");
 }
 
+inline u64 read_rndrrs()
+{
+    u64 value = 0;
+
+    asm volatile(
+        "retry:\n"
+        "mrs %[value], s3_3_c2_c4_1 \n" // encoded RNDRRS register
+        "b.eq retry\n"
+        : [value] "=r"(value));
+
+    return value;
+}
+
+inline u64 read_rndr()
+{
+    u64 value = 0;
+
+    asm volatile(
+        "retry:\n"
+        "mrs %[value], s3_3_c2_c4_0 \n" // encoded RNDR register
+        "b.eq retry\n"
+        : [value] "=r"(value));
+
+    return value;
+}
```

Comment:
This function is unused?

Reply:
yes; i don’t see it’s immediate usage but it is free to use, e.g. to produce more randomness/entropy etc.

not sure how safe it is and if those those snippets are to be used only during bootstrapping… 🤔

your call?

Reply:
Well if we currently don't use it (or this PR will not use it), we shouldn't add it either.

Reply:
added for completeness and symmetry only — will remove then! 👌